src/workload.rs

```diff
@@ -257,8 +259,26 @@ impl WorkloadManager {
         config: config::Config,
         metrics: Arc<Metrics>,
         awaiting_ready: readiness::BlockReady,
+        cert_manager: Box<dyn CertificateProvider>,
     ) -> anyhow::Result<WorkloadManager> {
-        let workloads: Arc<Mutex<WorkloadStore>> = Arc::new(Mutex::new(WorkloadStore::default()));
+        let (tx, mut rx) = watch::channel(Identity::default());
+        tokio::spawn(async move {
+            while rx.changed().await.is_ok() {
```

Comment:
We should add a TODO comment to rate limit these. OK for now though. I am looking into what the limit should be -- I think we definitely do not want ~255 requests at once, but not sure what is reasonable.